services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/BaseSpecification.groovy

```diff
@@ -50,24 +47,31 @@ class BaseSpecification extends SpringSpecification implements SetupOnce {
     }
 
     @HealthCheck
-    def "Given hardware topology is suitable for running the test suite"() {
-        requireProfiles("hardware")
+    def "Kilda is UP and topology is clean"() {
+        expect: "Kilda's health check request is successful"
+        northbound.getHealthCheck().components["kafka"] == "operational"
 
-        expect: "Topology and overall kilda state are suitable for running further test suite"
+        and: "All switches and links are active"
         def links = northbound.getAllLinks()
         verifyAll {
             northbound.activeSwitches.size() == topologyDefinition.activeSwitches.size()
-            northbound.allFlows.empty
-            northbound.allLinkProps.empty
+            links.findAll { it.state == IslChangeType.FAILED }.empty
             links.findAll {
                 it.state == IslChangeType.DISCOVERED
             }.size() == topologyDefinition.islsForActiveSwitches.size() * 2
-            links.findAll { it.state == IslChangeType.FAILED }.empty
         }
 
-        and: "More detailed inspection also reports no issues"
+        and: "No flows and link props are present"
```

Comment:
It's fine, but it is not the same as it was before and I don't find it to be an improvement. Previously it would report you about all problems at once (except for second round of verifications which are not very typical to fail). Right now you will have to fix links, then rerun, then fix flows, then rerun again.

Reply:
Fixed.